src/main/java/us/kbase/auth2/providers/OrcIDIdentityProviderFactory.java

```diff
@@ -228,6 +233,58 @@ private OrcIDAccessTokenResponse(
 				this.accessToken = accessToken.trim();
 				this.fullName = fullName == null ? null : fullName.trim();
 				this.orcID = orcID.trim();
+				this.idToken = idToken == null ? null : idToken.trim();
+				this.mfaAuthenticated = parseAmrClaim(this.idToken);
+			}
+			
+			/**
+			 * Parses the Authentication Method Reference (AMR) claim from an OpenID Connect ID token
+			 * to determine if multi-factor authentication was used.
+			 * 
+			 * @param idToken the JWT ID token from ORCID
+			 * @return true if MFA was used, false if password only, null if unknown or parsing failed
```

Comment:
We should definitely not conflate a provider not supporting mfa and a mfa processing error